core/trino-parser/src/main/java/io/trino/sql/tree/ArithmeticBinaryExpression.java

```diff
@@ -45,6 +45,7 @@ public String getValue()
     private final Operator operator;
     private final Expression left;
     private final Expression right;
+    private int hash;
```

Comment:
See (`Node`):
```
    // Force subclasses to have a proper equals and hashcode implementation
    @Override
    public abstract int hashCode();
```
Maybe we should rename to `hashCode` to `computeHashCode` and have abstract hash code caching at `Node` abstract class level? wdyt @martint @findepi ?

Reply:
This would force to move also equals() to Node level (+ some implementation in each class), as maven checkstyle plugin will complain about class having implementation of equals but missing hashCode.